chain/network/src/peer_manager/peer_store/mod.rs

```diff
@@ -588,6 +597,46 @@ impl PeerStore {
         self.0.lock().add_peer(clock, peer_info, TrustLevel::Direct)
     }
 
+    /// Accepts a time-ordered list of recently connected peers and merges it with the store's
+    /// list of recent connections, evicting the oldest peers if MAX_TIER2_PEERS is reached.
```

Comment:
we should evict peers which are disconnected for the longest time. Please document this fact and add a test for this behavior.